independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -407,6 +410,29 @@ ValidationContext validate(List<BeanDeploymentValidator> validators,
         return Collections.unmodifiableList(beans);
     }
 
+    public Collection<BeanInfo> getBeansByType(Type type) {
+        Map<Type, List<BeanInfo>> byType = beansByType;
+        if (byType == null) {
+            synchronized (this) {
+                byType = beansByType;
+                if (byType == null) {
+                    byType = new HashMap<>();
+                    for (var i : beans) {
+                        for (var t : i.types) {
```

Comment:
`var i`, `var t`... I find this style very unreadable and a potential source of bugs from maintenance POV.